drake/systems/onlinePlanning/OnlinePlannerMex.m

```diff
@@ -0,0 +1,152 @@
+classdef OnlinePlannerMex < DrakeSystem
+    
+    properties
+        % Constant properties
+        funnelLibrary;
+        funnelLibraryReduced;
+        t_replan;
+        cyclicIdx;
+        
+        % Mutable properties
+        currentFunnel;
+        forestAhead;
+        t_start;
+        x_start;
+        init_replan;
+        
+    end
+    
+    methods
+        
+        % construction
+        function obj = OnlinePlannerMex(funnelLibrary,forestAhead,currentFunnel,cyclicIdx,t_replan,t_start,x_start,num_inputs)
```

Comment:
forestAhead is pretty problem specific, when most other things in your class are pretty generic.